src/java/org/apache/cassandra/repair/autorepair/AutoRepairUtils.java

```diff
@@ -443,6 +462,252 @@ public static AutoRepairHistory getHostWithLongestUnrepairTime(RepairType repair
         return getHostWithLongestUnrepairTime(autoRepairHistories);
     }
 
+    /**
+     * Convenience method to resolve the broadcast address of a host id
+     *
+     * @return broadcast address if it exists, otherwise null.
+     */
+    @Nullable
+    private static InetAddressAndPort getBroadcastAddress(UUID hostId)
+    {
+        return StorageService.instance.getEndpointForHostId(hostId);
```

Comment:
I think this returns the `peer_address` instead of `rpc_address` so it may not work if they don't match (generally they should match)

Reply:
Good catch! Let me take a look at it.

Reply:
I looked further into it and found that in `AutoRepairUtils.java` the replica calculation also uses peer_address, so wouldn't it even out? In short, since at all places inside `AutoRepairUtils.java` we are using `peer_address`, it should be fine. 

```
            AbstractReplicationStrategy replicationStrategy = entry.getKey();
            EndpointsByRange endpointsByRange = replicationStrategy.getRangeAddresses(StorageService.instance.getTokenMetadata());

            // get ranges of the eligible address for the given replication strategy.
            RangesAtEndpoint rangesAtEndpoint = StorageService.instance.getReplicas(replicationStrategy, eligibleBroadcastAddress);

```